### PR TITLE
[FormRecognizer] Attributing Cognitive Services User role to test resources

### LINKED ISSUE #13126: [Live Test Failure] Form Recognizer AAD tests failing
See: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=442681&view=ms.vss-test-web.build-test-results-tab

Enable `FormRecognizerClientCanAuthenticateWithTokenCredential` and `FormTrainingClientCanAuthenticateWithTokenCredential` once issue has been solved.

### PR DESCRIPTION
**Pull request by kinelski**
AAD tests started to fail because Cognitive Services User role was not being attributed to the test resource anymore (it was being attributed by default somehow). Updating the `test-resources.json` file so we can explicitly specify that this role must attributed to the created Form Recognizer resource.

Fixes #13126.

### CODE REVIEW

File: sdk/formrecognizer/test-resources.json [hunk 1]
```diff
@@ -60,6 +66,7 @@
         }
     },
     "variables": {
+        "roleDefinitionId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/a97b65f3-24c7-4388-baec-2e87135dc908')]",
```
Comment (kinelski) [inline sdk/formrecognizer/test-resources.json:69]: ID (a97b65f3-24c7-4388-baec-2e87135dc908) obtained from: https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles

**Review by maririos — APPROVED** [review]

**Pull request merged**